Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealitySearchUtility.cs

```diff
@@ -0,0 +1,225 @@
+                }
+
+                // Check to see if this is a profile - if it is, enter the profile and check its children
+                bool isProfileField = false;
+                if (iterator.propertyType == SerializedPropertyType.ObjectReference && iterator.objectReferenceValue != null)
+                {   // If the property is itself a profile, perform a recursive search
+                    Type referenceType = iterator.objectReferenceValue.GetType();
+                    if (typeof(BaseMixedRealityProfile).IsAssignableFrom(referenceType))
+                    {
+                        isProfileField = true;
+                        foreach (ProfileSearchResult subResult in SearchProfileFields(iterator.objectReferenceValue, config))
+                        {
+                            yield return subResult;
+                        }
+                    }
+                }
+
+                // Gather our top-level info
+                bool keywordReqsMet = !config.RequireAllKeywords;
+                int fieldMatchStrength = 0;
+
+                CheckFieldKeywords(config, iterator, ref fieldMatchStrength, ref keywordReqsMet);
+                if (keywordReqsMet)
+                {
+                    if (ProfileSearchResult.IsEmpty(result))
+                    {
+                        result.Profile = profile;
+                        BaseMixedRealityProfile baseProfile = (profile as BaseMixedRealityProfile);
+                        result.IsCustomProfile = (baseProfile != null) ? baseProfile.IsCustomProfile : false;
+                        result.ProfileMatchStrength = fieldMatchStrength;
+                        result.Fields = new List<FieldSearchResult>();
+                    }
+
+                    result.MaxFieldMatchStrength = Mathf.Max(result.MaxFieldMatchStrength, fieldMatchStrength);
+                    result.Fields.Add(
+                        new FieldSearchResult()
+                        {
+                            Property = iterator.Copy(),
+                            MatchStrength = fieldMatchStrength,
+                        });
+                }
+
+                if (config.SearchChildProperties)
+                {
+                    // If this property has children, we want to search those children for relevant results
+                    // But we don't want to exit our current level
+                    if (!isProfileField && iterator.hasChildren && (iterator.isArray || iterator.propertyType == SerializedPropertyType.Generic))
+                    {
+                        if (!serializedPropertyTypesToFlatten.Contains(iterator.type))
+                        {
+                            SerializedProperty childIterator = iterator.Copy();
+                            bool hasChildProperties = childIterator.NextVisible(true);
+                            int depthOnEnter = childIterator.depth;
+                            while (hasChildProperties)
```

Comment:
what are children of a SerializedProperty?

Also when searching the children, we are only looking for profiles?

Reply:
Array values, members in a struct, etc.